merge dz4 to master

3RUN opened the pull request:
![Screenshot_1561320726](https://user-images.githubusercontent.com/3964596/59981488-aac01600-960c-11e9-8903-8fc7ca342931.png)

3RUN committed:
empty project

3RUN committed:
first upload

3RUN committed:
added lines

3RUN committed:
visual fixes

tim-teacher commented on app/src/main/java/by/itacademy/pvt/dz4/Dz4ClockView.kt:
```diff
@@ -0,0 +1,164 @@
+
+        canvas.drawCircle(clockCircleX, clockCircleY, clockCenterDotRadius, clockCenterDotPaint)
+
+        postInvalidateDelayed(500)
```
хитро :)

tim-teacher commented on app/src/main/java/by/itacademy/pvt/dz4/Dz4ClockView.kt:
```diff
@@ -0,0 +1,164 @@
+                clockLinesPaint.strokeWidth = resources.getDimension(R.dimen.Dz4ClockLineMainWidth)
+                clockLinesPaint.color = ContextCompat.getColor(context, R.color.dz3White)
+            } else {
+                sizeOffset = resources.getDimension(R.dimen.Dz4ClockLinesLength)
```
чтение из ресурсов трудоемкая операция, лучше вынести это в init и имя с маленькой буквы пишите - dz4ClockLinesLength

tim-teacher commented on app/src/main/java/by/itacademy/pvt/dz4/Dz4ClockView.kt:
```diff
@@ -0,0 +1,164 @@
+                clockLinesPaint.color = ContextCompat.getColor(context, R.color.dz3Gray)
+            }
+            val startPosY = (screenCenterHeight - clockCircleRadius)
+            val endPosY = (screenCenterHeight - clockCircleRadius) + sizeOffset
```
вобще sizeOffset я бы рекомендовал как то рассчитывать, а не задавать значением

tim-teacher reviewed (approved):
Все ок, только вынесите чтение из ресурсов в init и куда то что бы не вызывался каждый раз в onDraw()

3RUN committed:
added fixes:
- migrated resources into init
- dimen names with lowercase (dz4 and dz3)
- dz2 layout fix
